evalutils/io.py

```diff
@@ -63,6 +81,50 @@ def load(self, *, fname: Path) -> List[Dict]:
             If you reference a Path it must be saved in the `path` column
 
 
+        Parameters
+        ----------
+        fname
+            The file that the loader will try to load
+
+        Returns
+        -------
+            A list containing all of the cases in this file
+
+        Raises
+        ------
+        FileLoaderError
+            If a file cannot be loaded as the specified type
+
+        """
+        if self._include_pattern:
+            if not re.match(self._include_pattern, str(fname)):
+                raise FileLoaderIncludePatternError(
+                    f"File does not match include pattern "
+                    f"(reg.exp.): {self._include_pattern}"
+                )
+        if self._exclude_pattern:
```

Comment:
I don't think that this should be done here. This changes what the `load` method should be doing, from just loading a file to `filter_and_load`. I know that client code overrides this so try to think of another approach that does not break that. Maybe add a new function `_filter_files` and call it with the globbed list in `_load_cases`.

Reply:
It indeed does `filter_and_load` instead of just `load`. You might be right that some clients have already written classes overriding `load`. I might better just add a `file_filters` option to the BaseProcess class then with type `dict[str, Pattern[str]]`.

Reply:
I have moved the filtering to the BaseProcess class.